app/src/test/java/com/google/android/ground/ui/tos/TermsOfServiceFragmentTest.kt

```diff
@@ -63,60 +56,67 @@ class TermsOfServiceFragmentTest : BaseHiltTest() {
-        return Html.toHtml(SpannableStringBuilder(textView.text), 0) == html
-      }
-
-      override fun describeMismatch(item: Any?, description: Description?) {
-        description?.appendText(Html.toHtml(SpannableStringBuilder((item as TextView).text), 0))
-        super.describeMismatch(item, description)
-      }
-    }
+  /**
+   * composeTestRule has to be created in the specific test file in order to access the required
+   * activity. [composeTestRule.activity]
+   */
+  @get:Rule override val composeTestRule = createAndroidComposeRule<ComponentActivity>()
 
   override fun setUp() {
     super.setUp()
     fakeRemoteDataStore.termsOfService = Result.success(TEST_TOS)
   }
 
+  @Test
+  fun `Toolbar is displayed`() {
+    launchFragmentInHiltContainer<TermsOfServiceFragment>(bundleOf(Pair("isViewOnly", false)))
+
+    composeTestRule
+      .onNodeWithText(composeTestRule.activity.getString(R.string.tos_title))
+      .assertIsDisplayed()
+  }
+
+  @Test
+  fun `Toolbar Back Arrow is displayed`() {
+    launchFragmentInHiltContainer<TermsOfServiceFragment>(bundleOf(Pair("isViewOnly", true)))
+
+    composeTestRule.onNodeWithContentDescription("Back").assertIsDisplayed()
+  }
+
   @Test
   fun termsOfServiceText_shouldBeDisplayed() {
     whenever(networkManager.isNetworkConnected()).thenReturn(true)
     launchFragmentInHiltContainer<TermsOfServiceFragment>(bundleOf(Pair("isViewOnly", false)))
 
-    onView(withId(R.id.termsText))
-      .check(matches(isDisplayed()))
-      .check(matches(withText("This is a heading\n\nSample terms of service\n\n")))
-      .check(
-        matches(
-          withHtml(
-            "<p dir=\"ltr\"><span style=\"font-size:1.50em;\"><b>This is a heading</b></span></p>\n" +
-              "<p dir=\"ltr\">Sample terms of service</p>\n"
-          )
-        )
-      )
+    composeTestRule
+      .onNodeWithText("This is a heading\n\nSample terms of service\n\n")
+      .assertExists()
   }
 
   @Test
   fun agreeButton_default_shouldNotBeEnabled() {
     launchFragmentInHiltContainer<TermsOfServiceFragment>(bundleOf(Pair("isViewOnly", false)))
 
-    onView(withId(R.id.agreeCheckBox)).check(matches(isNotChecked()))
-    onView(withId(R.id.agreeButton)).check(matches(not(isEnabled())))
+    composeTestRule
+      .onNodeWithText(composeTestRule.activity.getString(R.string.agree_checkbox))
+      .assertIsDisplayed()
+
+    composeTestRule
+      .onNodeWithText(composeTestRule.activity.getString(R.string.agree_terms))
+      .assertIsNotEnabled()
   }
 
   @Test
   fun agreeButton_whenCheckBoxClicked_shouldBeEnabled() {
     launchFragmentInHiltContainer<TermsOfServiceFragment>(bundleOf(Pair("isViewOnly", false)))
 
-    onView(withId(R.id.agreeCheckBox)).perform(click()).check(matches(isChecked()))
-    onView(withId(R.id.agreeButton)).check(matches(isEnabled()))
+    composeTestRule
+      .onNodeWithText(composeTestRule.activity.getString(R.string.agree_checkbox))
```

Comment:
Let's create helper methods for getting checkbox and button nodes and reuse everywhere?